### PR TITLE
[OpenAI] Added missing GA service versions, 2024-02-01 and 2024-06-01

### PR DESCRIPTION
**Pull request by mssfang**
We have announced a GA release version, `2024-02-01`, but it is not supported in the [latest TypeSpec 's service version list](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/OpenAI.Inference/main.tsp#L57). 

We have this GA version in the repo.
https://github.com/Azure/azure-rest-api-specs/tree/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/stable/2024-02-01

Updates (2024/07/10):
We just have `2024-06-01` Service API version supported recently. This PR will add it to the Service API version list as well.

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/main.tsp [hunk 1]
```diff
@@ -53,6 +53,9 @@ enum ServiceApiVersions {
   @useDependency(Azure.Core.Versions.v1_0_Preview_1)
   v2023_07_01_Preview: "2023-07-01-preview",
 
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  v2024_02_01: "2024-02-01",
+
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/main.tsp:58]: Per the TypeSpec meeting conversation, this looks great -- can we also add `2024-06-01` (which just went live!)?

Comment (mssfang) [reply specification/cognitiveservices/OpenAI.Inference/main.tsp:58]: will do

**Review by trrwilson — APPROVED** [review]

File: specification/ai/Face/tspconfig.yaml [hunk 2]
```diff
@@ -37,6 +37,7 @@ options:
     package-dir: "azure-ai-vision-face"
     namespace: com.azure.ai.vision.face
     partial-update: true
+    stream-style-serialization: false
```
Comment (mssfang) [inline specification/ai/Face/tspconfig.yaml:40]:
```suggestion
```

**Comment by timotheeguerin** [comment]
This pr was merged with typespec build failing…

**Pull request merged**